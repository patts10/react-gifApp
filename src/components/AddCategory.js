import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

  const [inputvalue, setInputvalue] = useState("");

  const handleInputChange = (e) => {
    setInputvalue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputvalue.trim().length > 2) {
      setCategories((c) => [inputvalue, ...c]);
      setInputvalue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        className="my-5"
        type="text"
        value={inputvalue}
        onChange={handleInputChange}
        placeholder={"Search..."}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
