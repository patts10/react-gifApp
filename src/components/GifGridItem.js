import React from 'react';

export const GifGridItem = ( {title, url }) => {


  return (
    <div className='overflow-hidden rounded-md border-black border-2 xs:w-1/2 md:w-2/12 animate__animated animate__fadeIn mx-px my-5'>
      <img 
        className='w-full md:h-5/6 sm:h-5/6 xs:h-5/6'
        src={ url }
        alt={ title }
      />
      <p className='my-1 text-center'>{ title }</p>
    </div>
  );
};
