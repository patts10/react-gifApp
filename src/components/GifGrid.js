import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category );

  

  return (
    <div className="bg-slate-100 p-5">
      <h3 className="text-lg font-bold text-gray-900 sm:text-3xl sm:truncate mt-3 mb-5 mt-5"> { category } </h3>

      { loading && <p className="animate__animated animate__flash">Loading...</p>}

      <div>
        <ol className="flex flex-wrap justify-around">
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </ol>
      </div>
    </div>
  );
};