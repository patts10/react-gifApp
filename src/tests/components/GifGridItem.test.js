
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests on <GifGridItem />', () => {
  
  const title = 'tittle';
  const url = 'https://localhost/something.jpg'
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  
  test('should show component correctly', () => {
    
    expect(wrapper).toMatchSnapshot();
  });

  test('should exist a paragraph with the title', () => {
    
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });
  
  test('should exist an image with its url and alt that they are similar to props', () => {
    
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });
  
  test('should have animate__fadeIn class', () => {
    
    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true);
  });
  
});
