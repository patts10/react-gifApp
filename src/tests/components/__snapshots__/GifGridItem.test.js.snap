// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> should show component correctly 1`] = `
<div
  className="overflow-hidden rounded-md border-black border-2 xs:w-1/2 md:w-2/12 animate__animated animate__fadeIn mx-px my-5"
>
  <img
    alt="tittle"
    className="w-full md:h-5/6 sm:h-5/6 xs:h-5/6"
    src="https://localhost/something.jpg"
  />
  <p
    className="my-1 text-center"
  >
    tittle
  </p>
</div>
`;
